controllers/datamovement_controller.go

```diff
@@ -423,11 +423,12 @@ func getCmdAndArgs(cmCommand string, cmNumProcesses int, progressInterval int, h
 
 		cmd = "mpirun"
 		args = []string{
-			"--allow-run-as-root", // TODO: UserId/GroupId
+			"--allow-run-as-root", // required to be able to set dcp uid/gid
 			"-np", fmt.Sprintf("%d", numProcesses),
 			"--host", strings.Join(hosts, ","),
 			"dcp",
 			"--progress", fmt.Sprintf("%d", progressInterval),
+			fmt.Sprintf("-U %d -G %d", dm.Spec.UserId, dm.Spec.GroupId),
```

Comment:
I thought arguments couldn't be merged as a single string

Reply:
Strange. Well, you're inconsistent with --progress. You added long-form options, right? I always prefer those for clarity